contracts/src/v0.8/dev/ocr2/OCR2BaseUpgradeable.sol

```diff
@@ -0,0 +1,397 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../ConfirmedOwnerUpgradeable.sol";
+import "./OCR2Abstract.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/**
+ * @notice Onchain verification of reports from the offchain reporting protocol
+ * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
+ * @dev For details on its operation, see the offchain reporting protocol design
+ * doc, which refers to this contract as simply the "contract".
+ * @dev This contract is meant to aid rapid development of new applications based on OCR2.
+ * However, for actual production contracts, it is expected that most of the logic of this contract
+ * will be folded directly into the application contract. Inheritance prevents us from doing lots
+ * of juicy storage layout optimizations, leading to a substantial increase in gas cost.
+ */
+abstract contract OCR2BaseUpgradeable is Initializable, ConfirmedOwnerUpgradeable, OCR2Abstract {
```

Comment:
All of these inherited contracts have their own state, if any were reordered in future versions it would cause storage collisions.